### ISSUE TITLE
🌳 Improving Trees Code

### ISSUE THREAD

**Issue by dpshelio** [UCL-COMP0233-24-25/RSE-Classwork#39]
[exercise-repo]: https://github.com/UCL-COMP0233-24-25/refactoring-examples/tree/main/trees

Open the [repository for this exercise][exercise-repo]. You will find a brief description of each example in its `README.md` file.

1. Discuss the code with your group. What problems do you see with it? Why?
2. Write below your comments mentioning your group number. For now, focus on the
   problems, and don't make any changes yet!

**Comment by iaheung** [comment]
**Group 4:**
- The calculations inside the append function can be written to a subfunction/helper function
- It is not clear what the variables are attributed to
- The variable names can be more explicit

**Comment by D1Vincenzo** [comment]
Group 7

1. Add some docstrings and comments
2.  Change the variables name to be more clear
3. Overall improve the readibility

**Comment by r4b61t** [comment]
### Group 5 
- Has 1 comment that is completely useless
- Variables' names don't convey what it represents clearly
- calculations should be encapsulated in a function

**Comment by azure80** [comment]
Group 1:
- Variable names are unhelpful, make it hard to follow what is happening.
- No indication as to what the indexing relates to
- So specified use-case for the trees code
- Lack comments and docstrings

**Comment by YangMM24** [comment]
Group 2

- Missing comment
- The loop structure is not encapsulated as a function
- Variables are not clearly named
- It might be better to use variables instead of hardcoding (e.g. 0.1)

**Comment by zcemksd** [comment]
Group 6:

- There are not enough comments, it's not obvious what the code is doing 
- Name variables are not clear: what are 's', 'd', 'n'? 
- The calculations could be made into a function that works outside the loop and is called in the loop to help with clarity of the code understanding
